sdk/src/Services/DynamoDBv2/Custom/DataModel/_bcl/Context.Sync.cs

```diff
@@ -299,118 +299,92 @@ public void ExecuteTransactWrite(params TransactWrite[] transactionParts)
-        /// <returns>Lazy-loaded collection of results.</returns>
+        /// <inheritdoc/>
         public IEnumerable<T> Scan<T>(params ScanCondition[] conditions)
         {
-            return Scan<T>(conditions, null);
+            return Scan<T>(conditions, (ScanConfig)null);
         }
 
-        /// <summary>
-        /// Executes a Scan operation against DynamoDB, finding items
-        /// that match the specified conditions.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="conditions">
-        /// Conditions that the results should meet.
-        /// </param>
-        /// <param name="operationConfig">Config object which can be used to override that table used.</param>
-        /// <returns>Lazy-loaded collection of results.</returns>
+        /// <inheritdoc/>
+        [Obsolete("Use the Scan overload that takes ScanConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to Scan.")]
         public IEnumerable<T> Scan<T>(IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig)
         {
             var scan = ConvertScan<T>(conditions, operationConfig);
             return FromSearch<T>(scan);
         }
 
-        /// <summary>
-        /// Executes a Scan operation against DynamoDB, finding items
-        /// that match the specified conditions.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="scanConfig">Scan request object.</param>
-        /// <param name="operationConfig">Config object which can be used to override the table used.</param>
-        /// <returns>Lazy-loaded collection of results.</returns>
-        public IEnumerable<T> FromScan<T>(ScanOperationConfig scanConfig, DynamoDBOperationConfig operationConfig = null)
+        /// <inheritdoc/>
+        public IEnumerable<T> Scan<T>(IEnumerable<ScanCondition> conditions, ScanConfig scanConfig)
+        {
+            var scan = ConvertScan<T>(conditions, scanConfig?.ToDynamoDBOperationConfig());
+            return FromSearch<T>(scan);
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<T> FromScan<T>(ScanOperationConfig scanOperationConfig)
         {
-            if (scanConfig == null) throw new ArgumentNullException("scanConfig");
+            if (scanOperationConfig == null) throw new ArgumentNullException("scanOperationConfig");
 
-            var search = ConvertFromScan<T>(scanConfig, operationConfig);
+            var search = ConvertFromScan<T>(scanOperationConfig, null);
+            return FromSearch<T>(search);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Use the FromScan overload that takes ScanConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to FromScan.")]
+        public IEnumerable<T> FromScan<T>(ScanOperationConfig scanOperationConfig, DynamoDBOperationConfig operationConfig = null)
```

Comment:
Renaming the argument from `scanConfig` -> `scanOperationConfig` would be a breaking change